api/graphql/types/reservations/serializers/adjust_time_serializers.py

```diff
@@ -32,7 +31,16 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.fields["state"].readonly = True
 
-    def save(self, **kwargs):
+    def save(self) -> Reservation:
+        kwargs: dict[str, Any] = {}
+        for res_unit in self.instance.reservation_unit.all():
```

Comment:
When user adjusts the time of the reservation, the buffers should always be re-calculated according to the current reservation unit settings (whether or not blocking the whole day), just like when creating a new reservation.

Reply:
OK, is this a new rule, since I don't think reservations currently do this? I suppose there isn't much harm in it?

Reply:
The only place I see this as being meaningful is when creating new reservations. Let's say that users make reservations with one buffer, which is then changed, and afterward, more users begin making reservations. The reservable time slots appear shorter or longer depending on whether the previous reservations users have edited their reservations or not.

Reply:
As it is/was, these two scenarios would give different end results for customer users:
A:
1. User makes reservation (say Monday 12-13.00 on date X) to a res.unit with 30 min buffer settings.
2. Res.unit settings are changed and the buffer lengths are increased to 60 mins. (Any existing reservations, including the above, still have their 30 min individual buffers, of course.)
3. User cancels the reservation, and then makes a new one for the following Thursday (date X + 2) where there is a free slot; the new one has the 60 min buffer settings applied to it.

B: same as A.1-2 plus:
3. Rather than cancelling the reservation, the user adjusts the time to the Thursday; it keeps the 30 buffers that the reservation originally had.

But that doesn't make sense since time adjustment is kind of shorthand for cancel & re-reserve, and by now it is known that the staff needs an hour to do whatever is needed for setup etc.
So the old logic of keeping the previously applied buffer times (whether zero or not), unless the res.unit "blocks the whole day", must be wrong.

Reply:
That example is true, but there are still issues with buffers not being applied unless the user edits the reservation. The implication is that there is an "activation date" for the new buffer, even if it's "right now", and that all reservations after that time conform to the new buffer.

If new reservations should always conform to the reservation unit's buffer, what's the point of moving the buffer to the reservation?

Reply:
One reason I can think of is admin reservations, where the buffer can be overridden. Perhaps there should be "buffer overrides" on the reservations, rather than always including a buffer?

Reply:
Anyways, removed the condition to skip updating buffers, so now every reservation will get new buffers on time adjust.

Reply:
That's a fair point; admin reservations are probably the main reason. But secondarily, once a reservation exists and is not subsequently moved, it may be desirable to "lock" its buffers, too, and to be able to make fixed plans according to those.
Say a reservation has a 60 min buffer for cleanup afterwards, but the res.unit setting is later decreased to 30 mins for new reservations. But that specific cleanup in the real life may already have been scheduled, so don't want to allow another reservation *now* to be made only 30 mins after it.

(Whereas if a reservation is moved to another time altogether, all bets are off and I can't think of why its original buffers would have relevance over the current res.unit settings. I think that was just not accounted for when this serializer was written.)

Reply:
Things to think about for the reservations refactor. I guess we can merge this and think about the edge case later?